<head>
  <meta name="author" content="Rogério de Oliveira">
  <meta institution="author" content="Universidade Presbiteriana Mackenzie">
</head>

<img src="http://meusite.mackenzie.br/rogerio/mackenzie_logo/UPM.2_horizontal_vermelho.jpg" width=300, align="right">
<!-- <h1 align=left><font size = 6, style="color:rgb(200,0,0)"> optional title </font></h1> -->

**CURSO DE PÓS-GRADUAÇÃO EM CIÊNCIA DE DADOS (BIG DATA PROCESSING AND ANALYTICS)**<p>
**Componente curricular:** BIG DATA ANALYTICS: ANÁLISE ESTATÍSTICA - 91162 - 2022.2 - 01D - Trilha 2.<br>

**Aluno:** ROBSON DE FREITAS SAMPAIO.<br>

**URL deste notebook:** https://github.com/rfsampaio/postgraduate_data_science/blob/main/notebooks/AE_T2.ipynb<br>
***

### **Atividade - Trilha 2**

Nesta atividade, você utilizará o R/RStudio para resolver alguns problemas simples envolvendo Distribuições de Probabilidade; também fará alguns testes de hipóteses e resolverá problemas de Inferência. Começamos com alguns exercícios que deverão ser resolvidos no R.

1. Suponha que o peso médio dos Pinguins Reis encontrados em uma colônia na Antártica no último ano foi 15.4 kg. Em uma amostra de 35 pinguins na mesma época neste ano, na mesma colônia, o peso médio foi 14.1kg. Assuma que o desvio padrão da população seja 2kg. A um nível de significância de 5%, nós podemos rejeitar a hipótese nula de que o peso médio dos pinguins não difere do último ano?

2. Na indústria cerâmica, avalia-se sistematicamente a resistência de amostras de massas cerâmicas, após o processo de queima. Dessas avaliações, sabe-se que certo tipo de massa tem resistência mecânica aproximadamente normal, com média 53 MPa e variância 16 MPa<sup>2</sup>. Após a troca de alguns fornecedores de matérias-primas, deseja-se verificar se houve alteração na qualidade. Uma amostra de 15 corpos de prova de massa cerâmica acusou média igual a 50 MPa. Qual é a conclusão ao nível de significância de 5 %?

3. O tempo para transmitir 10 MB em determinada rede de computadores varia segundo um modelo normal, com média 7,4 seg e variância 1,3 seg<sup>2</sup>. Depois de algumas mudanças na rede acredita-se numa redução no tempo de transmissão de dados, além de uma possível alteração na variabilidade. Foram realizados 10 ensaios independentes com um arquivo de 10 MB e foram anotados os tempos de transmissão, em segundos: 6,8; 7,1; 5,9; 7,5; 6,3; 6,9; 7,2; 7,6; 6,6; 6,3

    Existe evidência suficiente de que o tempo médio de transmissão foi reduzido? Use nível de significância de 1%.

### **Carregando bibliotecas necessárias:**

In [2]:
library(tidyverse)

### **Resolução:**

**Exercício 1**

#### $ z_{observado} = {(\bar{x}-\mu) \over {\sigma \over \sqrt{n}}} $<br>
$ \bar{x} = 14.1 $<br>
$ \mu = 15.4 $<br>
$ \sigma = 2.0 $<br>
$ n = 35 $<br>
<br>
$ H_{0} = 15.4 $ kg<br>
$ H_{1} \neq 15.4 $ kg<br>
$ \alpha = 5.0 $ %<br>
$ z_{observado} = z_{crítico} \to $ Aceita $ H_{0} $.<br>
$ z_{observado} \neq z_{crítico} \to $ Rejeita $ H_{0} $.<br>

**Cálculos no R da estatística de teste e dos valores de "z" para o Exercício 1:**

In [3]:
# população
mu <- 15.4; cat('mu =', mu, '\n') # média da população
sigma <- 2.0; cat('sigma =', sigma, '\n') # desvio padrão da população
alpha <- 0.05; cat('alpha =', alpha, '\n') # nível de significância bilateral de 5%

# amostra
xbar <- 14.1; cat('xbar =', xbar, '\n') # média da amostra
n <- 35 # tamanho da amostra

# cálculo dos Zs
zo <- (xbar - mu) / (sigma / sqrt(n)); cat('zo =', zo, '\n') # z observado
zc <- qnorm(1 - alpha / 2); cat('zc =', zc, '\n') # z crítico

# região de aceitação
ra <- c(-zc, zc); cat('ra =', ra, '\n')

# resultado do teste
res_aceita_h0 <- zo == zc; cat('Aceita H0?', res_aceita_h0, '\n') # aceita H0
res_rejeita_h0 <- zo != zc; cat('Rejeita H0?', res_rejeita_h0) # rejeita H0


mu = 15.4 
sigma = 2 
alpha = 0.05 
xbar = 14.1 
zo = -3.845452 
zc = 1.959964 
ra = -1.959964 1.959964 
Aceita H0? FALSE 
Rejeita H0? TRUE

**Conclusão do Exercício 1:**

A estatística de teste zo = -3.84 está fora da região de aceitação, compreendida entre -1.96 e 1.96. Assim, de acordo com os dados coletados e adotando um nível de significância de 5%, podemos rejeitar a hipótese nula de que o peso médio dos pinguins não difere do último ano.

**Exercício 2**

#### $ z_{observado} = {(\bar{x}-\mu) \over {\sigma \over \sqrt{n}}} $<br>
$ \bar{x} = 50.0 $<br>
$ \mu = 53.0 $<br>
$ \sigma = \sqrt{16} $<br>
$ n = 15 $<br>
<br>
$ H_{0} = 53.0 $ MPa<br>
$ H_{1} \neq 53.0 $ MPa<br>
$ \alpha = 5.0 $ %<br>
$ z_{observado} = z_{crítico} \to $ Aceita $ H_{0} $.<br>
$ z_{observado} \neq z_{crítico} \to $ Rejeita $ H_{0} $.<br>

**Cálculos no R da estatística de teste e dos valores de "z" para o Exercício 2:**

In [4]:
# população
mu <- 53.0; cat('mu =', mu, '\n') # média da população
sigma <- sqrt(16); cat('sigma =', sigma, '\n') # desvio padrão da população
alpha <- 0.05; cat('alpha =', alpha, '\n') # nível de significância bilateral de 5%

# amostra
xbar <- 50.0; cat('xbar =', xbar, '\n') # média da amostra
n <- 15 # tamanho da amostra

# cálculo dos Zs
zo <- (xbar - mu) / (sigma / sqrt(n)); cat('zo =', zo, '\n') # z observado
zc <- qnorm(1 - alpha / 2); cat('zc =', zc, '\n') # z crítico

# região de aceitação
ra <- c(-zc, zc); cat('ra =', ra, '\n')

# resultado do teste
res_aceita_h0 <- zo == zc; cat('Aceita H0?', res_aceita_h0, '\n') # aceita H0
res_rejeita_h0 <- zo != zc; cat('Rejeita H0?', res_rejeita_h0) # rejeita H0

mu = 53 
sigma = 4 
alpha = 0.05 
xbar = 50 
zo = -2.904738 
zc = 1.959964 
ra = -1.959964 1.959964 
Aceita H0? FALSE 
Rejeita H0? TRUE

**Conclusão do Exercício 2:**

A estatística de teste zo = -2.90 está fora da região de aceitação, compreendida entre -1.96 e 1.96. Assim, de acordo com os dados coletados e adotando um nível de significância de 5%, podemos rejeitar a hipótese nula de que não houve alteração na qualidade.

**Exercício 3**

Como a variância da população é desconhecida deveremos utilizar a distribuição "t" de Student para calcular a estatística de teste. Vide abaixo:
<br>
#### $ t_{observado} = {(\bar{x}-\mu) * \sqrt{n} \over {\sigma}} $<br>
$ \bar{x} = ? $<br>
$ \mu = 7.4 $<br>
$ \sigma = ? $<br>
$ n = 10 $<br><br>
Considerando que a variabilidade possa ter sido alterada, vamos usar a variância da amostra. Fazendo os cálculos com as 10 observações, teremos que obter a média, a variância e o desvio padrão da amostra. Conjunto de observações: 6,8; 7,1; 5,9; 7,5; 6,3; 6,9; 7,2; 7,6; 6,6; 6,3. O grau de liberdade "gl" será dado pela quantidade de observações - 1, ou seja, 10 - 1 = 9.
<br><br>
$ H_{0} = 7.4 $ s<br>
$ H_{1} < 7.4 $ s<br>
$ \alpha = 1.0 $ %<br>
$ t_{observado} = t_{crítico} \to $ Aceita $ H_{0} $.<br>
$ t_{observado} < t_{crítico} \to $ Rejeita $ H_{0} $.<br>

**Cálculos no R da estatística de teste e dos valores de "t" para o Exercício 3:**

In [5]:
# população
mu <- 7.4; cat('mu =', mu, '\n') # média da população
alpha <- 0.01; cat('alpha =', alpha, '\n') # nível de significância unilateral de 1%

# amostra
xbar <- mean(c(6.8, 7.1, 5.9, 7.5, 6.3, 6.9, 7.2, 7.6, 6.6, 6.3))
cat('xbar =', xbar, '\n') # média da amostra
S <- var(c(6.8, 7.1, 5.9, 7.5, 6.3, 6.9, 7.2, 7.6, 6.6, 6.3))
cat('S =', S, '\n') # variância da amostra
sigma <- sqrt(S); cat('sigma =', sigma, '\n') # desvio padrão da amostra
n <- 10 # tamanho da amostra
gl <- n - 1 # grau de liberdade da amostra

# cálculo dos Ts
to <- (((xbar - mu) * sqrt(n)) / sigma); cat('to =', to, '\n') # t observado
tc <- qt(alpha, df = gl); cat('tc =', tc, '\n') # t crítico

# resultado do teste
res_aceita_h0 <- to == tc; cat('Aceita H0?', res_aceita_h0, '\n') # aceita H0
res_rejeita_h0 <- to < tc; cat('Rejeita H0?', res_rejeita_h0) # rejeita H0

mu = 7.4 
alpha = 0.01 
xbar = 6.82 
S = 0.304 
sigma = 0.551362 
to = -3.326528 
tc = -2.821438 
Aceita H0? FALSE 
Rejeita H0? TRUE

**Conclusão do Exercício 3:**

A estatística de teste to = -3.33 está fora da região de aceitação, pois é menor que -2.82. Assim, de acordo com os dados coletados e adotando um nível de significância de 1%, podemos rejeitar a hipótese nula de que o tempo médio de transmissão não foi reduzido.